server/adka2a/metadata.go

```diff
@@ -103,3 +135,38 @@ func setActionsMeta(meta map[string]any, actions session.EventActions) map[strin
 	}
 	return meta
 }
+
+func processA2AMeta(a2aEvent a2a.Event, event *session.Event) error {
+	taskInfo, meta := a2aEvent.TaskInfo(), a2aEvent.Meta()
+
+	if gm, ok := meta[metadataGroundingKey].(map[string]any); ok {
```

Comment:
Let's add the test case when there's `nil` value for metadataGroundingKey. To make sure the code handles this (not sure how artificial this case is).

If this makes sense, let's add this check for all added fields `metadataGroundingKey`, `metadataUsageKey` etc.